arangod/Aql/Ast.cpp

```diff
@@ -1888,6 +1911,10 @@ void Ast::injectBindParameters(
             }
           }
           node = newNode;
+
+          if (cachedNode == nullptr) {
+            parameters.registerNode(param, node);
+          }
```

Comment:
What if `cacheNode != nullptr`? Couldn't we also just lookup the param and use a previously registered datasource node instead?

Reply:
That _may_ be true, but I am unaware of the side effects for this type of node.
At least it does no harm to always create a new AstNode in this case, as the AstNode will always have a small size (in contrast to arbitrarily-sized bind parameter values from JSON/VPACK).
I added more comments to the code to make the intentions more clear.

Reply:
Ok... but the newly created nodes should be equivalent to any cached ones. Should we at least add some assertions to check that?

Reply:
Added assertions in latest commit.